core/controllers/reader.py

```diff
@@ -169,6 +169,95 @@ def classify(state, answer):
         'exploration owner.')
 
 
+class ExplorationPageEmbed(base.BaseHandler):
+    """Page describing a single embeded exploration."""
+
+    @require_playable
+    def get(self, exploration_id):
+        """Handles GET requests."""
+        version_str = self.request.get('v')
+        version = int(version_str) if version_str else None
+
+        # Note: this is an optional argument and will be None when the
+        # exploration is being played outside the context of a collection.
+        collection_id = self.request.get('collection_id')
+
+        try:
```

Comment:
OK, so -- we talked about refactoring to minimize code duplication. Here's how you can do it:

- Create a top-level function in this file called _get_exploration_player_data(exploration_id, version, collection_id). It returns a dict with the GADGET_SPECS, INTERACTION_SPECS, etc. -- pretty much everything except 'iframed', basically.
- Make both this controller and the ExplorationPage controller call this top-level function.
- Augment the return dict with the value for 'iframed' and pass it to self.values.update, then render the correct template.

Let's see how things look after this, but I think this should reduce a lot of the duplication here.